app/src/hooks/useSwap.ts

```diff
@@ -9,142 +9,272 @@ import React from 'react'
-        const updatedValue = (Number(value) / rate).toFixed(6).toString()
-        setToAmount(updatedValue)
+      if (selectedNetwork?.coin === CoinType.Solana) {
+        await handleJupiterQuoteRefresh({
+          fromAmount: value
+        })
+      } else {
+        await handleZeroExQuoteRefresh({
+          fromAmount: value,
+          toAmount: ''
+        })
       }
     },
-    [toToken, rate]
+    [selectedNetwork, handleJupiterQuoteRefresh, handleZeroExQuoteRefresh]
   )
 
-  const handleOnSetToAmount = React.useCallback((value: string) => {
-    setToAmount(value)
-  }, [])
+  const handleOnSetToAmount = React.useCallback(
+    async (value: string) => {
+      // Setting to amount is not supported on Jupiter
+      if (selectedNetwork?.coin !== CoinType.Ethereum) {
+        return
+      }
+
+      setToAmount(value)
+      await handleZeroExQuoteRefresh({
+        fromAmount: '',
+        toAmount: value
+      })
+    },
+    [selectedNetwork, handleZeroExQuoteRefresh]
+  )
 
   const getTokenBalance = React.useCallback(
     (token: BlockchainToken): string => {
-      return tokenBalances ? tokenBalances[token.contractAddress] ?? '0' : '0'
+      return tokenBalances
+        ? new Amount(tokenBalances[token.contractAddress] ?? '0')
+          .divideByDecimals(token.decimals)
+          .format(6)
+        : '0'
     },
     [tokenBalances]
   )
 
   const onClickFlipSwapTokens = React.useCallback(() => {
     setFromToken(toToken)
     setToToken(fromToken)
-  }, [toToken, fromToken])
+    setFromAmount('')
+    setToAmount('')
+  }, [fromToken, toToken])
 
-  const onSelectToToken = React.useCallback((token: BlockchainToken) => {
-    setToToken(token)
-    setSelectingFromOrTo(undefined)
-  }, [])
+  const onSelectToToken = React.useCallback(
+    async (token: BlockchainToken) => {
+      setToToken(token)
+      if (selectedNetwork?.coin === CoinType.Solana) {
+        await handleJupiterQuoteRefresh({
+          toToken: token
+        })
+      } else if (selectedNetwork?.coin === CoinType.Ethereum) {
+        await handleZeroExQuoteRefresh({
+          toToken: token,
+          toAmount: ''
+        })
+      }
 
-  const onSelectFromToken = React.useCallback((token: BlockchainToken) => {
+      setSelectingFromOrTo(undefined)
+    },
+    [selectedNetwork, handleJupiterQuoteRefresh]
+  )
+
+  const onSelectFromToken = React.useCallback(async (token: BlockchainToken) => {
     setFromToken(token)
-    setSelectingFromOrTo(undefined)
-  }, [])
+    if (selectedNetwork?.coin === CoinType.Solana) {
+      await handleJupiterQuoteRefresh({
+        fromToken: token
+      })
+    } else if (selectedNetwork?.coin === CoinType.Ethereum) {
+      await handleZeroExQuoteRefresh({
+        fromToken: token,
+        fromAmount: ''
+      })
+    }
 
-  const onSelectQuoteOption = React.useCallback((option: QuoteOption) => {
-    setSelectedQuoteOption(option)
+    setSelectingFromOrTo(undefined)
   }, [])
```

Comment:
Missing `selectedNetwork`, `handleJupiterQuoteRefresh` and `handleZeroExQuoteRefresh` dependancies.